models/classes/class.LtiBasicOutcome.php

```diff
@@ -156,4 +158,9 @@ public function storeItemVariables($deliveryResultIdentifier, $test, $item, arra
             $this->storeItemVariable($deliveryResultIdentifier, $test, $item, $itemVariable, $callIdItem);
         }
     }
+
+    private function getLtiOutcomeXmlFactory(): LtiOutcomeXmlFactory
+    {
+        $this->getServiceLocator()->get(LtiOutcomeXmlFactory::class);
```

Comment:
I think that is the must here
```suggestion
       return $this->getServiceLocator()->get(LtiOutcomeXmlFactory::class);
```